src/__tests__/send-request.js

```diff
@@ -1,4 +1,72 @@
-import { encodePostData } from '../send-request'
+import { encodePostData, xhr } from '../send-request'
+
+const generateXmlHttpRequestMock = (status = 502) => ({
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    send: jest.fn(),
+    readyState: 4,
+    responseText: JSON.stringify('something here'),
+    status: status,
+})
+
+const generateXhrParams = (markRequestFailed, onXHRError = () => {}) => ({
+    url: 'https://any.posthog-instance.com',
+    data: '',
+    headers: {},
+    options: {},
+    captureMetrics: {
+        incr: () => {},
+        startRequest: () => {},
+        decr: () => {},
+        finishRequest: () => {},
+        markRequestFailed,
+    },
+    callback: () => {},
+    retriesPerformedSoFar: null,
+    retryQueue: {
+        enqueue: () => {},
+    },
+    onXHRError,
+})
+
+describe('when xhr requests fail', () => {
+    given('mockXHR', generateXmlHttpRequestMock)
```

Comment:
Nit: I think this function could be inlined here to make it a bit easier to follow what's happening here (one less step in the trail)